src/engraving/dom/scorefile.cpp

```diff
@@ -115,17 +116,31 @@ bool Score::loadStyle(const String& fn, bool ign, const bool overlap)
     TRACEFUNC;
 
     File f(fn);
-    if (f.open(IODevice::ReadOnly)) {
-        MStyle st = style();
-        if (st.read(&f, ign)) {
-            undo(new ChangeStyle(this, st, overlap));
-            return true;
-        } else {
-            LOGE() << "The style file is not compatible with this version of MuseScore Studio.";
-            return false;
-        }
+    return doLoadStyle(f, ign, overlap);
+}
+
+bool Score::loadStyle(const muse::ByteArray& data, bool ign, bool overlap)
+{
+    TRACEFUNC;
+
+    muse::io::Buffer buffer(data.constData(), data.size());
```

Comment:
This call will result in unnecessary copying of data (you can look at the implementation)

Reply:
```
    ByteArray ba = ByteArray::fromRawData(data.constData(), data.size());
    Buffer buf(&ba);
```